aws_topology/stackstate_checks/aws_topology/aws_topology.py

```diff
@@ -161,17 +181,18 @@ def get_topology(self, instance_info, aws_client):
                     }
                     self.event(event)
                     errors.append("API %s ended with exception: %s %s" % (spec["api"], str(e), traceback.format_exc()))
-            # TODO this should be for tests, in production these relations should not be sent out
-            agent_proxy.finalize_account_topology()
-            self.components_seen = agent_proxy.components_seen
-            if len(errors) > 0:
-                raise Exception("get_topology gave following exceptions: %s" % ", ".join(errors))
+        # TODO this should be for tests, in production these relations should not be sent out
+        self.log.info('Finalize FULL scan (#components = {})'.format(len(agent_proxy.components_seen)))
+        agent_proxy.finalize_account_topology()
+        self.components_seen = agent_proxy.components_seen
         self.delete_ids += agent_proxy.delete_ids
 
         self.stop_snapshot()
+        if len(errors) > 0:
```

Comment:
I can't remember if we decided to actually make this change where the errors are emitted after the snapshot - does it make any difference?

Reply:
I think this makes more sense

Reply:
Broken record :smile: You really shouldn't do this. It will cause flickering topology. Let's figure out why we're currently seeing the flickering, fix it and then keep the pattern that is used across all integrations